tests/func/test_external_repo.py

```diff
@@ -125,56 +125,36 @@ def test_relative_remote(erepo_dir, tmp_dir):
             assert fd.read() == "contents"
 
 
-def test_shallow_clone_branch(erepo_dir):
-    with erepo_dir.chdir():
-        with erepo_dir.branch("branch", new=True):
-            erepo_dir.dvc_gen("file", "branch", commit="create file on branch")
-        erepo_dir.dvc_gen("file", "master", commit="create file on master")
-
-    url = os.fspath(erepo_dir)
-
-    with patch.object(Git, "clone", wraps=Git.clone) as mock_clone:
-        with external_repo(url, rev="branch") as repo:
-            with repo.open_by_relpath("file") as fd:
-                assert fd.read() == "branch"
-
-        mock_clone.assert_called_with(url, ANY, shallow_branch="branch")
-        _, shallow = CLONES[url]
-        assert shallow
-
-        with external_repo(url) as repo:
-            with repo.open_by_relpath("file") as fd:
-                assert fd.read() == "master"
-
-        assert mock_clone.call_count == 1
-        _, shallow = CLONES[url]
-        assert not shallow
-
-
-def test_shallow_clone_tag(erepo_dir):
-    with erepo_dir.chdir():
-        erepo_dir.dvc_gen("file", "foo", commit="init")
-        erepo_dir.scm.tag("v1")
-        erepo_dir.dvc_gen("file", "bar", commit="update file")
-
-    url = os.fspath(erepo_dir)
-
-    with patch.object(Git, "clone", wraps=Git.clone) as mock_clone:
-        with external_repo(url, rev="v1") as repo:
-            with repo.open_by_relpath("file") as fd:
-                assert fd.read() == "foo"
-
-        mock_clone.assert_called_with(url, ANY, shallow_branch="v1")
-        _, shallow = CLONES[url]
-        assert shallow
-
-        with external_repo(url, rev="master") as repo:
-            with repo.open_by_relpath("file") as fd:
-                assert fd.read() == "bar"
-
-        assert mock_clone.call_count == 1
-        _, shallow = CLONES[url]
-        assert not shallow
+def test_shallow_clone(mocker):
```

Comment:
placeholder for now, may need to set up a docker container w/git server and fixed repo data for testing against different git server protocols?